### ISSUE TITLE
PortAudio Windows Error and Compatibility on Python3.7 

### ISSUE THREAD

**Issue by Yagna24** [juanmc2005/StreamingSpeakerDiarization#28]
Hi, @juanmc2005  

Is there a chance that diart can run on windows since PortAudio library is only supported on Linux ? 

Also, if its possible to import Literal from typing_extensions(py3.7)  instead of Typing(py3.8) 

I get this error while running this demo script  : 
```
import rx
import rx.operators as ops
import diart.operators as myops
from diart.sources import MicrophoneAudioSource
import diart.functional as fn

sample_rate = 16000
mic = MicrophoneAudioSource(sample_rate)
```

error : 
`ImportError: cannot import name 'Literal' from 'typing' (/usr/lib/python3.7/typing.py)`

**Comment by juanmc2005** [comment]
Hi @Yagna24,

As far as I understand, both portaudio and sounddevice are cross-platform and run on Windows: [sounddevice docs](https://python-sounddevice.readthedocs.io/en/0.4.4/), [portaudio docs](http://portaudio.com/docs/v19-doxydocs/tutorial_start.html).

Concerning `typing`, diart is compatible with python >= 3.8. Is there a particular reason you need to use python 3.7?

**Comment by Yagna24** [comment]
Yes. I have been trying to run the script on colab. 

I managed to install portaudio on colab simply by using !apt-get  install 

Also, currently colab supports py3.7, which is the reason why I need to use python 3.7.

**Comment by juanmc2005** [comment]
Ok, this looks like a nice feature to have then.
I'll try to add compatibility with python 3.7 but right now I don't have much free time.
If you are up for it, I would be happy to merge a PR.

**Comment by Yagna24** [comment]
Hi, I managed the 3.7 part. I cloned the repo made a change then pip installed from the repo directly to have it compatible with 3.7.

**Comment by juanmc2005** [comment]
Great! Would you mind opening a PR so I can include this in the next release?

Judging from the error you get I'm assuming that PortAudio is not finding your computer's microphone from colab. I'm not quite sure how to go around that, you may need to manually set the the device or to implement a custom AudioSource.

**Comment by Yagna24** [comment]
Yes, I could open a Pull request, The one change I made is import Litertal from typing_extensions instead of typing in functional.py . 

`from typing import Literal `--> Removed 
`from typing_extensions `--> added 

Also, to make PortAudio work on colab I installed via this command : 

`!apt install libasound2-dev portaudio19-dev libportaudio2 libportaudiocpp0 ffmpeg`

Are these changes fine for you ?

**Comment by juanmc2005** [comment]
I'm ok with those changes, you can create the PR and I'll make sure to take a look so we can include this feature

**Comment by Yagna24** [comment]
> I'm ok with those changes, you can create the PR and I'll make sure to take a look so we can include this feature

Sounds fair

**Comment by juanmc2005** [comment]
PR #29 merged
